src/components/button/_button.scss

```diff
@@ -80,14 +81,26 @@
       border: 0;
     }
 
-    &:hover,
     &:focus {
+      // !important is used to ensure that when the user hovers over a focussed button
+      // it does not change background/Text colour
+      color: govuk-colour("black") !important;
+      background-color: govuk-colour("yellow") !important;
+      box-shadow: 0 4px 0 govuk-colour("black");
+    }
+
+    &:hover {
+      color: govuk-colour("white");
       background-color: $govuk-button-hover-colour;
     }
 
     &:active {
-      top: $button-shadow-size;
-      box-shadow: none;
+
+      border: $govuk-border-width-form-element solid govuk-colour("yellow");
+      // !important is used to overwrite the :focus !important colour
+      color: govuk-colour("white") !important;
+      background-color: $govuk-button-hover-colour !important;
```

Comment:
Are you sure we need to put these !important statements in? I wonder if these pseudo selectors could be rearranged to avoid this.

Reply:
Good spot 👍